src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,254 @@
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteInlineEditableCancelEditing: EventEmitter;
+
+  @Event() calciteInlineEditableConfirmChanges: EventEmitter;
+
+  @Event() calciteInlineEditableEnableEditing: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInputBlur")
+  blurHandler(): void {
+    if (!this.hasControls) this.disableEditing();
+  }
+
+  @Listen("click", { target: "window" })
+  handleLabelFocus(e: CustomEvent): void {
+    const HTMLTarget = e.target as HTMLElement;
+    if (!HTMLTarget.classList.contains("calcite-label-text")) return;
+    if (!HTMLTarget.parentElement.contains(this.el)) return;
+    e.preventDefault();
+    e.stopPropagation();
+    this.enableEditingButton.setFocus();
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private State
+  //
+  //--------------------------------------------------------------------------
+
+  private inputElement: HTMLCalciteInputElement;
+
+  private htmlInput: HTMLInputElement;
+
+  private valuePriorToEditing: string;
+
+  private enableEditingButton: HTMLCalciteButtonElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private enableEditing = () => {
+    this.htmlInput.tabIndex = undefined;
+    this.valuePriorToEditing = this.inputElement.value;
+    this.editingEnabled = true;
+    this.inputElement.setFocus();
+    this.calciteInlineEditableEnableEditing.emit();
+  };
+
+  private disableEditing = () => {
+    this.htmlInput.tabIndex = -1;
+    this.editingEnabled = false;
+  };
+
+  private cancelEditing = () => {
+    this.inputElement.value = this.valuePriorToEditing;
+    this.disableEditing();
+    setTimeout(() => this.enableEditingButton.setFocus(), 100);
+    this.calciteInlineEditableCancelEditing.emit();
+  };
+
+  private escapeKeyHandler = async (e: KeyboardEvent) => {
+    if (e.key !== "Escape") return;
+    this.cancelEditing();
+  };
+
+  private cancelEditingHandler = async (e: MouseEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.cancelEditing();
+  };
+
+  private enableEditingHandler = async (e: MouseEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (!this.editingEnabled) this.enableEditing();
+  };
+
+  private confirmChangesChangesHandler = async (e: MouseEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    try {
+      if (this.onConfirmChanges) {
+        this.loading = true;
+        await this.onConfirmChanges();
+        this.disableEditing();
+      }
+      this.calciteInlineEditableConfirmChanges.emit();
```

Comment:
If you're emitting this `calciteInlineEditableConfirmChanges` event, providing the `this.onConfirmChanges` would be redundant because the consumer can just listen for the `calciteInlineEditableConfirmChanges` event like so:

```
<calcite-inline-editable onCalciteInlineEditableConfirmChanges={() => {}}>
```

Unless I'm missing something else about the need for this extra callback function...